### ISSUE TITLE
Build pip requirements.txt to specific runtime (cross macos-linux/architecture compilation)

### ISSUE THREAD

**Issue by caioquirino** [terraform-aws-modules/terraform-aws-lambda#459]
## Is your request related to a problem? Please describe.

I am having to use docker to build a lambda layer from a pip requirements.txt file, but since I am doing it locally from a macbook, the binaries are added for darwin architecture. 

It would be great if I can force it to use linux/amd64 (manylinux) or something that is compatible with AWS Lambda when building it locally

## Describe the solution you'd like.

It would be great if together with `pip_requirements` in the `source_path` list, I could use something like `platform="linux/amd64"` or even following the pypa manylinux standards as `platform="manylinux2014_x86_64"`.

Or as another option, to use the `compatible_runtimes` and `architectures` or `runtime` fields already existing in the module.

## Describe alternatives you've considered.

If I can fix the issue https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/458 that would be another alternative

**Comment by caioquirino** [comment]
Up

**Comment by caioquirino** [comment]
Up...

**Comment by jcopps** [comment]
Up..

**Comment by atyndall** [comment]
Ran into this, would be very useful

**Comment by jcopps** [comment]
### Two workaround options:
 @atyndall , @caioquirino 

1. One may try this: https://stackoverflow.com/a/74729370/3195195
2. Use docker with pip requirements.txt. In this case, the docker can be set with the right platform architecture and the build takes care of itself.

**Comment by atyndall** [comment]
Thanks @jcopps. It would be great if the module could set the `--platform manylinux2014_x86_64` itself.

**Comment by jcopps** [comment]
Agreed. But the docker support was added for a reason like these I presume.

**Comment by caioquirino** [comment]
@jcopps it exists, but it is not working for my use case (cross compiling mac-os m1 to linux x64): https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/458

**Comment by caioquirino** [comment]
@jcopps you can find a lot of information about what is happening in the #458 issue :)
TLDR: Yes

**Comment by userhas404d** [comment]
I ran into this recently as a result of [this](https://github.com/pyca/cryptography/issues/6391) issue. I was able to cheese it locally by manually cramming in the pip options required [here](https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/package.py#L983) but needed _both_ `--platform` and `--only-binary`, as a result my recommendation for this issue (and solution) would be to allow the input of an arbitrary number of additional `pip` options

**Comment by atyndall** [comment]
Not stale
